src/utils/bump.js

```diff
@@ -0,0 +1,163 @@
+'use strict'
+
+const semver = require('semver')
+
+async function getAutoBumpedVersion({ github, context }) {
+  const { owner, repo } = context.repo
+
+  const {
+    latestReleaseCommitSha,
+    latestReleaseTagName,
+    latestReleaseCommitDate,
+  } = await getLatestRelease({ github, owner, repo })
+
+  if (
+    !latestReleaseCommitSha ||
+    !latestReleaseTagName ||
+    !latestReleaseCommitDate
+  ) {
+    throw new Error(`Couldn't find latest release`)
+  }
+
+  const allCommits = await getCommitMessagesSinceLatestRelease({
+    github,
+    owner,
+    repo,
+    commitDate: latestReleaseCommitDate,
+  })
+
+  if (!allCommits.length) {
+    throw new Error(`No commits found since last release`)
+  }
+
+  const bumpedVersion = getVersionFromCommits(latestReleaseTagName, allCommits)
+
+  if (!semver.valid(bumpedVersion)) {
+    throw new Error(`Invalid bumped version ${bumpedVersion}`)
+  }
+  return bumpedVersion
+}
+
+function getVersionFromCommits(currentVersion, commits = []) {
+  // Define a regular expression to match Conventional Commits messages
+  const commitRegex = /^(feat|fix|BREAKING CHANGE)(\(.+\))?:(.+)$/
```

Comment:
let's use an official conventionalcommits package to parse this and implement all this logic please!

Reply:
Ok I was thinking about it as well but then decided to keep it simple. They have a conventionalcommits parser lib so I will just use that